conan/tools/microsoft/nmaketoolchain.py

```diff
@@ -38,7 +38,9 @@ def _format_defines(self, defines):
                 if value and not value.isnumeric():
                     value = f'\\"{value}\\"'
                 define = f"{macro}#{value}"
-            formated_defines.append(f"/D{define}")
+                if " " in define:
+                    define = f"\"{define}\""
+            formated_defines.append(f'/D{define}')
```

Comment:
Why this different than ``NMakeDeps``? can't we use the same approach (always quoting)?

Reply:
I have removed the conditional to the whitespace, as it works quoting always. However, it seems that the string values need an extra quote pair. I have tested it at the nmake toolchain side and is the only combination that works. Removing the non-numeric conditional throws the following errors:

```
simple.cpp(101): error C2065: 'TestPpdValue4': undeclared identifier
E           simple.cpp(101): error C2059: syntax error: '__declspec(dllexport)'
E           simple.cpp(101): error C2065: 'foo': undeclared identifier
E           simple.cpp(101): error C2146: syntax error: missing ';' before identifier 'bar'
E           simple.cpp(101): error C2065: 'bar': undeclared identifier
```